### PR TITLE
langchain-mongodb: Remove in-memory cache from cache abstractions

### PR DESCRIPTION
**Pull request by Jibola**
## Description
* In memory cache easily gets out of sync with the server cache, so we will remove it entirely to reduce the issues around invalidated caches.

## Dependencies
None

- [x]  If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

### CODE REVIEW

**Comment by Jibola** [comment]
cc: @shaneharvey, @blink1073 , @NoahStapp , @caseyclements

**Comment by Jibola** [comment]
> What was the rationale for implementing a local cache?

Having the additional in-memory cache optimized on explicit replica prompts. Unfortunately, things get out of alignment between local and atlas more often than we liked.

File: libs/partners/mongodb/tests/unit_tests/test_cache.py [hunk 1]
```diff
@@ -30,7 +30,6 @@ def __init__(
         self.__database_name = database_name
```
Comment (ShaneHarvey) [inline libs/partners/mongodb/tests/unit_tests/test_cache.py:30]: Can we remove this file (test_cache.py) altogether?

Comment (Jibola) [reply libs/partners/mongodb/tests/unit_tests/test_cache.py:30]: We still need this file as it tests the `MongoDBAtlasSemanticCache` and `MongoDBCache` abstractions which both leverage MongoDB as the caching layer.

**Review by ShaneHarvey — APPROVED** [review]

**Comment by Jibola** [comment]
cc: @efriis

**Review by efriis — APPROVED** [review]

**Pull request merged**